service/consul/datacenterWatch.go

```diff
@@ -0,0 +1,109 @@
+package consul
+
+import (
+	"errors"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/service"
+)
+
+//DatacenterWatcher checks if datacenters have been updated, based on an interval
+//If they have, then the DatacenterWatcher will update the Instancers dictionary
+type DatacenterWatcher struct {
+	watchInterval time.Duration
+	logger        log.Logger
+	shutdown      chan struct{}
+	environment   Environment
+	options       Options
+}
+
+var (
+	defaultLogger = log.NewNopLogger()
+)
+
+func newDatacenterWatcher(logger log.Logger, environment Environment, options Options) (*DatacenterWatcher, error) {
```

Comment:
Is `newDatacenterWatcher()` meant to be unexported? No other package needs it?

Reply:
Hmmm yeah I left this as unexported because I envisioned that it was only going to be used when one creates a new consul service discovery environment. But I guess one could create a service discovery environment and then later create a datacenterWatcher without creating it in the `NewEnvironment` function. Although this would require whoever is using this datacenterWatcher to change the `DatacenterWatchInterval` duration from 0 to a number greater than 0 after it is passed into `NewEnvironment`, and I'm not sure why anybody would do that